common/src/result.ts

```diff
@@ -1,31 +1,19 @@
 import { sprintf } from 'sprintf-js';
```

Comment:
This one is a bit nit-picky but maybe result.ts should be results.ts since it contains multiple result types

Reply:
Result type is a union of the other types, the main type is Result, the rest are just helpers to create Result in different forms.